### DISCUSSION TITLE
Panels to record the travel distance

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#494] [Q&A] [answered]
Trip meter is a new panel that automatically measures the distance that has been traveled.

- Set to work in selected modes, e.g. navigation, follow mode

- **Reset** at any journey point to measure the distance traveled in any journey part
(like average speed - tap for menu)

- Works independently of GPS recording

- 2 trip meters are available in "Settings | Application | Display"

<img src="https://github.com/devemux86/cruiser/assets/3484020/263235e7-51b4-4842-af7a-ec3af3258619" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/cdc4526f-1648-4eef-9a52-0b404a37d370" width="50%"/>

**Comment by devemux86** [comment]
Is "Odometer" the best name to be understood by most users?

https://en.wikipedia.org/wiki/Odometer

**Reply by Leeb64** [reply]
In German:Tageskilometerzähler

**Reply by Ciutnik** [reply]
In Polish I suggest one word: `Drogomierz`

**Reply by 55wal** [reply]
For a resetable distance display perhaps "Trip" or "Trip meter" is better than "Odometer".

According to the description in following screenshot from above linked wikipedia page:

![grafik](https://github.com/devemux86/cruiser/assets/167004125/e8734778-0015-4c69-ad00-35432bc181f0)

**Reply by airald** [reply]
I think you are true. The odometer in vehicles indicates the total km count and it's no resettable. Partial km counter it's called usually trip.

**Reply by MotoUKRider** [reply]
Thinking about it, "Trip Counter" is what I refer to mine on the bike when I reset it after a fuel-stop.

**Comment by Cer-o** [comment]
I suppose it will be something planned for version 3.0.43, because I don't see it in the current one.

**Reply by devemux86** [reply]
@Cer-o it will be in the next version. Now it is in the Beta:

- https://github.com/devemux86/cruiser/discussions/482

**Comment by devemux86** [comment]
We can name it "Trip meter" if you like.

German seems to have the same translation "Tageskilometerzähler".
@Ciutnik Polish is "Licznik podróży"?

**Reply by Ciutnik** [reply]
As for the Polish translation: "Licznik podróży" (or "Licznik dzienny") = "Drogomierz". I consciously suggested the second one, because it's one word - it's shorter 🙂

It is true that "Trip counter" or "Daily counter" is clearly associated with a resettable counter. But in the application, probably no one expects that it will be a total mileage counter, which we have in vehicles 😁

**Comment by devemux86** [comment]
Great! Thank you very much for Odometer! Once, when I was planning a motorcycle orienteering rally, I would like to see two of them: one for individual stages (A), the other for the entire route (B) 🙂

But it seems that it counts raw GPS data, without filtering it. During a dozen or so minutes of stoppage, he showed a distance of 500 meters! True, the GPS signal was floating strongly.

So maybe, if possible, it would be better if the data were filtered, as is the case with "GPS recording"?

_Originally posted by @Ciutnik in https://github.com/devemux86/cruiser/discussions/482#discussioncomment-9114361._

**Reply by devemux86** [reply]
It is probably too much to introduce new settings for the trip meter?
We better use the filters of GPS recording?

About the number of meters, do you mean to manually reset them on each individual stage?
So have "Trip meter 1" and "Trip meter 2"?

**Reply by Ciutnik** [reply]
> We better use the filters of GPS recording?

This is exactly what I wanted to propose. No additional settings needed here!

> do you mean to manually reset them on each individual stage?

Yes. Not even so much in stages as in maneuvers! When designing the route in the field, I had to know the distance between turning points (roughly equivalent to the yellow dots on the Cruiser! 😁), to put the exact value on the printed itinerary. So the meter had to be reset several dozen times along the route 🙂

I determined the length of the route based on the saved GPX (this had the added benefit that when I hit a dead end or the concept changed, I cut out the unnecessary GPX sections before before calculating the total length). But a second resettable meter would be very useful then, because the trip meter on my motorcycle is reserved for the mileage between refueling.

It was great fun... But I'm sorry I went off topic.

> So have "Trip meter 1" and "Trip meter 2"?

Would be nice (1/2 or A/B if it would make it easier to identify on the panels?)👍

**Reply by devemux86** [reply]
> We better use the filters of GPS recording?

This will be used in the next update.

**Comment by devemux86** [comment]
We can have 2 trip meters that you can reset independently at any time.

Please see the updated images in the [1st post](https://github.com/devemux86/cruiser/discussions/494#discussion-6499211).

**Reply by bios4** [reply]
> that you can reset independently at any time

How ❓

When tapping the panel nothing happens.

And what exactly are these trip meters take into account when cumulating distances?
- Are they starting as soon as I "start navigation" or "start follow mode"?
- Are they automatically cumulating all distances recognized by the GPS sensor, independently of the mode I am in?
- ... ❔

And with a view to any future documentation we need a little more than just one/two sentences and a screenshot for such new features.

**Reply by devemux86** [reply]
Tap their panels to reset them each one independently.
(reset when they have already measured some distance)

They work when they are visible.
So if select to display them during navigation, they measure while navigation is active.

**Reply by devemux86** [reply]
We will need an improvement to make them _measure separately_.
Currently they measure together, not really what we want.

Coming in the next update. :slightly_smiling_face:

**Reply by bios4** [reply]
> Tap their panels to reset them each one independently.

Does that mean they would reset IMMEDIATELY when tapping?

Because when trying I was not "really" moving but only started navigation, so the panels showed 0 all the time - and tapping them "did nothing" at all...

My suggestion would be to ask for confirmation if they really should be reset - otherwise every accidential tap could lead to some loss of data/information.

**Reply by devemux86** [reply]
They show a reset question when they have distance, like when resetting the average speed.

**Reply by r11d11** [reply]
I paused navigation and after one minute trip meter was increased by 100m without movement  I think through gps inaccuracy. Would it be better, that the trip meter only counts in navigation?

**Reply by devemux86** [reply]
@r11d11 what have you selected in display settings of the 2 trip meters?
Just "Navigation" or also something else?

**Reply by r11d11** [reply]
I selected  Browse map for trip meter 2 and Navigation for trip meter 1. Now I understand. The Browse map setting leads to this behaviour. I thought trip meter works only in navigation and at the end of the day I could look in Browse map to the the whole km

**Reply by devemux86** [reply]
Trip meter (and average speed) work in a simple way: if their panels are visible, they measure.
GPS recording has specific start / stop actions and automatic recording in navigation & follow.

@r11d11 there was a fix in B7, now the 2 trip meters are really separated.

**Reply by Ciutnik** [reply]
It's true that meters in map view mode could be "read-only." Navigation and follow modes are for recording (counts)

**Reply by devemux86** [reply]
> It's true that meters in map view mode could be "read-only."

We could have it like this, but do we want it work that way?

Then we could not measure without navigation or follow, just by opening the app.

**Reply by Ciutnik** [reply]
> just by opening the app

This is also true. Hmm, hard to say. Each solution has its advantages. This feature is only one day old! Let's see what the needs of most users are 🙂

**Reply by devemux86** [reply]
> It's true that meters in map view mode could be "read-only." Navigation and follow modes are for recording (counts)

Let's do it this way, it is less confusing. So users don't suddenly see values increasing outside the 2 "moving" modes.
(this is also the current way in the release)

**Comment by devemux86** [comment]
> We will need an improvement to make them measure separately.

The improvement is implemented in the latest:
- https://github.com/devemux86/cruiser/discussions/482

**Comment by Ciutnik** [comment]
To maintain the grammatical form throughout the application, please change Polish `Reset licznik podróży`( _Reset trip meter_) to `Resetuj licznik podróży`. Thanks! 🙂

**Reply by devemux86** [reply]
@Ciutnik thanks, I will update it.

"Reset average speed" is "Reset średnia prędkość". Is this correct?

**Reply by Ciutnik** [reply]
Analogously, but with different ending (-ą): "Resetuj średnią prędkość". 
I didn't notice it! Thank you for drawing my attention 👍

God, why have you made my language so confusing?! 🤣

**Comment by devemux86** [comment, marked as answer]
Trip meter for measuring the distance traveled is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
